### PR TITLE
#150127619  Add Archive dialogue when archiving a contribution

### PR DESCRIPTION
**Pull request by kevintumbo**
#### What does this PR do?
- Adds archive button and dialog to component
#### Description of Task to be completed?
- Given that I am an admin
- When I click on the archive icon
- I should see confirmation dialogue when I want to archive a contribution
- Then the archive component should change to archived state.
#### How should this be manually tested?
After cloning the repo

- Run yarn install
- Run yarn start
- Access the page via http://calm-dev.andela.com:3000/contributions/archive and click on the folder icon
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#150127619](https://www.pivotaltracker.com/story/show/150127619)
#### Screenshots (if appropriate)
screen shot 2017-08-21 at 14 37 25

#### Questions:
N/A

### CODE REVIEW

**Comment by TUgonna** [comment]
@kevintumbo 
Check the engineering wiki guide and format your PR Title. Commit messages should be all lowercase

File: src/admin/Archive.component.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import './archive.scss';
+/**
+ * This class represents the ArchiveComponent class
+ * it  should change to archived state from false to true.
+ *
+ * @class ArchiveComponent
+ * @returns {any}
+ */
+export class ArchiveComponent extends React.Component <any, any> {
+
```
Comment (TUgonna) [inline src/admin/Archive.component.tsx]: Get rid of extra line

**Comment by andela-uofoegbu** [comment]
@kevintumbo
Check the engineering wiki guide and format your PR Title. Commit messages should be all lowercase

File: src/admin/Archive.component.tsx (hunk 1 above)
Comment (andela-uofoegbu) [inline src/admin/Archive.component.tsx]: remove extra line

File: src/admin/Archive.component.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  constructor(props) {
+      super(props);
```
Comment (andela-uofoegbu) [inline src/admin/Archive.component.tsx]: We indent with two spaces not four spaces

File: src/admin/Archive.component.tsx (hunk 1 above)
Comment (kevintumbo) [inline src/admin/Archive.component.tsx]: noted

File: src/admin/Archive.component.tsx (hunk 2 above)
Comment (kevintumbo) [inline src/admin/Archive.component.tsx]: noted

File: src/admin/archive.scss [hunk 3]
```diff
@@ -0,0 +1,4 @@
+.material-icons {
+    cursor: pointer;
```
Comment (andela-uofoegbu) [inline src/admin/archive.scss]: Indent with two spaces not four

File: src/admin/Archive.component.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+/**
+ * @memberof ArchiveComponent
+ * @returns actions
+ */
+  public getActions = () => {
+    const actions = [
+      { label: 'archive', onClick: this.archive},
+      { label: 'cancel', onClick: this.handleToggle},
+    ];
```
Comment (andela-uofoegbu) [inline src/admin/Archive.component.tsx]: May i know what handleToggle is supposed to do and what archive is supposed to do?

It seems they are doing the same thing. I would like some clarification

Comment (kevintumbo) [inline src/admin/Archive.component.tsx]: handleToggle changes the dialog's active state 
while archive changes the archived state from false to true

Comment (andela-uofoegbu) [inline src/admin/Archive.component.tsx]: toggleDialog is a better name for that function than handleToggle.

File: src/admin/Archive.component.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+/**
+ * @memberof ArchiveComponent
+ * @return tsx for archive icon and dialogue
+ */
+  public render() {
+    return (
+      <div id="archive">
+          <i onClick={this.handleToggle} className="material-icons">{
+            this.state.archived ? 'folder' : 'folder_open'}
```
Comment (andela-uofoegbu) [inline src/admin/Archive.component.tsx]: If a contribution is archived already? Should a dialog box display asking 
`<p> Are you sure you want to archive this Contribution? </p>`
I think the dialog box shouldn't have a static message. It should check if the contribution is already archived and display an appropriate message

**Pull request merged**